301-azure-firewall-dns-proxy/README.md

```diff
@@ -0,0 +1,794 @@
+    }
+  }
+]
+Checking if the prodvm file system already exists in the adlsvteqnecsplyqu storage account...
+The prodvm file system already exists in the adlsvteqnecsplyqu storage account
+Checking if the documents directory exists in the prodvm file system under the adlsvteqnecsplyqu storage account...
+The documents directory does not exist in the prodvm file system under the adlsvteqnecsplyqu storage account
+Creating the documents directory in the prodvm file system under the adlsvteqnecsplyqu storage account...
+{
+  "content_length": 0,
+  "continuation": null,
+  "date": "2021-03-15T08:39:50+00:00",
+  "etag": "\"0x8D8E78DE66C7513\"",
+  "last_modified": "2021-03-15T08:39:51+00:00",
+  "request_id": "bbc1db45-c01f-0050-4876-193b28000000",
+  "version": "2020-02-10"
+}
+The directoryName directory was successfully created in the prodvm file system under the adlsvteqnecsplyqu storage account
+Uploading the  file to the documents directory in the prodvm file system under the adlsvteqnecsplyqu storage account...
+{
+  "client_request_id": "0094fb82-856a-11eb-abe0-000d3a457072",
+  "content_length": 0,
+  "date": "2021-03-15T08:39:52+00:00",
+  "etag": "\"0x8D8E78DE7155610\"",
+  "last_modified": "2021-03-15T08:39:52+00:00",
+  "request_id": "77619041-301f-0009-7576-19bcab000000",
+  "version": "2020-02-10"
+}
+The  file was successfully uploaded to the documents directory in the prodvm file system under the adlsvteqnecsplyqu storage account
+Listing the files inside the documents directory in the prodvm file system under the adlsvteqnecsplyqu storage account...
+[
+  {
+    "contentLength": 76,
+    "etag": "0x8D8E78DE7155610",
+    "group": null,
+    "isDirectory": false,
+    "lastModified": "2021-03-15T08:39:52",
+    "name": "documents/prodvm.txt",
+    "owner": null,
+    "permissions": null
+  }
+]
+```
+
+## DNS Forwarder
+
+When the value of the `deployCustomDnsForwarder` parameter is set to `true`, a Linux virtual machine is deployed in the hub virtual network and the [Azure Custom Script Extension](https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-linux) is used to download and run the following Bash script on the virtual machine which installs and configure NGINX to forward DNS requests on port 53 to the default Azure DNS at `168.63.129.16`.
+
+```bash
+#!/bin/bash
+
+apt-get update -y && apt-get upgrade -y
+apt-get install -y nginx
+echo "Hello World from host" $HOSTNAME "!" | sudo tee -a /var/www/html/index.html
+
+touch /etc/nginx/nginx.conf
+cat >> /etc/nginx/nginx.conf <<EOF
+stream {
+      upstream dns_servers {
+       server 168.63.129.16:53;
+}
+
+server {
+ listen x.x.x.x:53  udp;
+ listen x.x.x.x:53; #tcp
+ proxy_pass dns_servers;
+ proxy_responses 1;
+ error_log  /var/log/nginx/dns.log info;
+}
+}
+EOF
+
+myip=`hostname -i | awk '{print $1}'`
+sed -i "s/x.x.x.x/$myip/" /etc/nginx/nginx.conf
+
+
+sudo nginx -t && sudo service nginx reload
+```
+
+## Deployment
```

Comment:
remove this section - deployment needs to be the same for all samples in the repo